docs/topics/data-analysis/data-analysis-work-with-data-sources.md

```diff
@@ -1,20 +1,124 @@
+
+## Display data
+
+Once you [have the data in your notebook](#retrieve-data-from-a-file), you can easily store it in a variable and access it by running the 
+following code cell: 
+
+```kotlin
+val dfJson = DataFrame.read("jsonFile.json")
+dfJson
+```
+
+This will display the data from the file of your choice, such as CSV, JSON, XLS, or XLSX.
+![Display data](display-data.png){width=700}
+
+To gain insights into the structure or schema of your data, apply the `.schema()` function on your DataFrame variable. 
+For example, `dfJson.schema()` will list the type of each column in your JSON dataset.
+
+![Schema example](schema-data-analysis.png){width=700}
+
+## Refine data
+
+Among the various functions available for refining your dataset in DataFrame, key examples include [grouping](https://kotlin.github.io/dataframe/group.html),
+[filtering](https://kotlin.github.io/dataframe/filter.html), [updating](https://kotlin.github.io/dataframe/update.html), 
+and [adding new columns](https://kotlin.github.io/dataframe/add.html). These functions are essential for data analysis, allowing you to organize, clean, and 
+transform your data effectively.  Let's look at an example where our data includes movie titles, which also contain 
+the release year of the movies. The goal is to refine this dataset for easier analysis:
+
+1. **Import libraries:** Import the DataFrame library necessary for data manipulation tasks.
+
+```kotlin
+%use dataframe
+```
+
+2. **Read data from a CSV file:**
+Load your data into the notebook. This example involves reading data from a CSV file named "movies.csv" and creating a DataFrame called `movies`:
+
+```kotlin
+val movies = DataFrame.read("movies.csv")
+```
+
+3. **Add a new data column for the year:** Extract the release year from the movie titles using regex and add it as a new column:
+
+```kotlin
+val moviesWithYear = movies
+    .add("year") { 
+        "\\d{4}".toRegex()
+            .findAll(title)
+            .lastOrNull()
+            ?.value
+            ?.toInt()
+            ?: -1
+    }
+```
+
+4. **Update movie titles:** Modify the movie titles by removing the release year from each title. 
+This cleans up the titles for consistency:
+
+```kotlin
+val moviesTitle = moviesWithYear
+    .update("title") {
+        "\\s*\\(\\d{4}\\)\\s*$".toRegex().replace(title, "")
+    }
+```
+
+5. **Filter movies and display the resulting DataFrame:** Use the `.filter` method to focus on specific data. 
+In this case, the dataset is filtered to focus on movies that were released after the year 1990:
+
+```kotlin
+val moviesNew = moviesWithYear.filter { year >= 1990 }
+moviesNew
+```
+
+![Data refinement result](refined-data.png){width=700}
+
+This is a practical demonstration of how DataFrame's functions, like `.add`, `.update`, and `.filter` can be used to 
+effectively refine and analyze data in Kotlin.
+
+For more examples, see [Examples of Kotlin Dataframe](https://github.com/Kotlin/dataframe/tree/master/examples). 
```

Comment:
Could we somehow describe that we have a writing support to all of these file formats? As a separate section or simple example like say, reading from csv -> remove a few columns -> filter data -> write to CSV or JSON

Reply:
👍  Of course, I'll try to come up with a way to integrate this info :)